Docker: Added Datadog support

rmanalan opened the pull request:
This PR adds support for the new Datadog integration as well as fixes some bugs I found when configuring the relay. The relay isn't required for this integration, just decided to fix it.

Commit:
Docker: Added Datadog supportI wanted to play with ld-relay using Support Service. This newdocker-compose adds support for that and dogstatsd for tracking cmetricson the ld-relay.

levlaz commented on docker-compose.yml:
```diff
@@ -1,38 +1,37 @@
-      - db
       - cache
 
-  db:
```
@rmanalan is there a specific reason why you got rid of the database here?

rmanalan replied:
Because I realized that the dev env doesn't require PG and will default to sqlite.

levlaz commented:
@rmanalan I think there are some mistakes here -- for instance did you intend to remove this file? 

> scripts/start_dev.sh

rmanalan commented:
> @rmanalan I think there are some mistakes here -- for instance did you intend to remove this file?
> 
> > scripts/start_dev.sh
Yeah, that was intentional since when I introduced this file, we needed to do a db migration and then run flask. All of that is done in the docker-compose.yml now.

levlaz commented:
@rmanalan thanks for the comments, this LGTM.

levlaz commented on requirements.txt:
```diff
@@ -24,3 +24,6 @@ launchdarkly-api==2.0.15
 gunicorn==19.9.0
+
+# Datadog
+datadog
```
@rmanalan can you please pin this to a specific version?

rmanalan replied:
Done.

levlaz commented:
Thanks @rmanalan this is good to go once https://github.com/launchdarkly/SupportService/pull/102 is merged.

Commit:
Merge branch 'master' into pr/101

Commit:
Added Datadog agent to default environment

This commit resorts to turning on the Datadog integration by default for all
environments. This will allow all LDers to demo the DD integration in context with
Support Service. I've enabled Datadog tracing in order to capture as
many metrics as possible without modifying code. The APM services can be
viewed at https://app.datadoghq.com/apm/services.

InTheCloudDan commented on .env.example:
```diff
@@ -19,3 +19,6 @@ LD_FRONTEND_KEY=
 # LD_API_KEY=
+
+# Uncomment the following section to send metrics to Datadog. This is wired to the Support Service (Dev) account.
+# DD_API_KEY=83821f1a517bc8eb9a6573e7fede9568
```
@rmanalan we need to rotate this key, we can add secrets into the circleci config, this is a public repo

rmanalan replied:
Ugh... sorry about that. I didn't realize this was public. Will rotate.

Commit:
fixup! Merge branch 'master' into pr/101

Commit:
fixup! Merge branch 'master' into pr/101

Commit:
fixup! Merge branch 'master' into pr/101

Commit:
fixup! Merge branch 'master' into pr/101

InTheCloudDan commented:
@levlaz is this good to go as far as you're concerned?
